Lib/asyncio/unix_events.py

```diff
@@ -51,8 +51,13 @@ def __init__(self, selector=None):
 
     def close(self):
         super().close()
-        for sig in list(self._signal_handlers):
-            self.remove_signal_handler(sig)
+        if not sys.is_finalizing():
+            for sig in list(self._signal_handlers):
+                self.remove_signal_handler(sig)
+        else:
+            warinigs.warn(f"Closing the loop {self!r} on interpreter shutdown "
+                          f"stage, signal unsubsription is disabled",
+                          ResourceWarning)
```

Comment:
For ResourceWarning, you can add source=self, so the warning will be logged with the traceback were the event loop was created, if traceback is enabled.

Reply:
Done